src/pmdas/bpf/modules/mountsnoop.c

```diff
@@ -0,0 +1,640 @@
+}
+
+static char* mountsnoop_metric_name(unsigned int metric)
+{
+    return metric_names[metric];
+}
+
+static unsigned int mountsnoop_indom_count(void)
+{
+    return INDOM_COUNT;
+}
+
+static void mountsnoop_set_indom_serial(unsigned int local_indom_id, unsigned int global_id)
+{
+    indom_id_mapping[local_indom_id] = global_id;
+}
+
+static int mountsnoop_metric_text(int item, int type, char **buffer)
+{
+    if (type & PM_TEXT_ONELINE) {
+        *buffer = metric_text_oneline[item];
+    } else {
+        *buffer = metric_text_long[item];
+    }
+
+    return 0;
+}
+
+static void mountsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmdaIndom *indoms)
+{
+    /* bpf.mountsnoop.pid */
+    metrics[PID] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 0),
+            .type  = PM_TYPE_U32,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.tid */
+    metrics[TID] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 1),
+            .type  = PM_TYPE_U32,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.comm */
+    metrics[COMM] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 2),
+            .type  = PM_TYPE_STRING,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.op */
+    metrics[OP] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 3),
+            .type  = PM_TYPE_STRING,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.ret */
+    metrics[RET] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 4),
+            .type  = PM_TYPE_STRING,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.lat */
+    metrics[LAT] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 5),
+            .type  = PM_TYPE_32,
+            .indom = indom_id_mapping[MOUNTSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
+        }
+    };
+    /* bpf.mountsnoop.mnt_ns */
+    metrics[MNT_NS] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 6),
+            .type  = PM_TYPE_U32,
```

Comment:
I think it this easy to change it to a string,
I used U32 because of this line:
https://github.com/iovisor/bcc/blob/0e29f7032897f2ebdd4c617910aa8ade74bc4b23/libbpf-tools/mountsnoop.h#L30